File: taller/AMarotoChivite-taller/AMarotoChivite-taller-Java/src/main/java/simulacionTaller/factories/ElectricianFactory.java
package simulacionTaller.factories;


import simulacionTaller.models.Electrician;
import simulacionTaller.models.Worker;

import static simulacionTaller.utils.randomInfo.*;

public class ElectricianFactory {
    public static Electrician create() {
        String name = randomName();
        int experienceYear = randomAnno();
        int salary = randomSalary();
        int dailyHours = randomDailyHours();

        return new Electrician(name, experienceYear, salary, dailyHours, Worker.BossAssigment.WITHOUT_BOSS);
    }
}

File: taller/AMarotoChivite-taller/AMarotoChivite-taller-kotlin/src/main/kotlin/simulacionTaller/factories/ElectricianFactory.kt
package simulacionTaller.factories

import simulacionTaller.models.Electrician
import simulacionTaller.models.Worker.BossAssigment
import simulacionTaller.utils.randomAnno
import simulacionTaller.utils.randomDailyHours
import simulacionTaller.utils.randomName
import simulacionTaller.utils.randomSalary

object ElectricianFactory {
    @JvmStatic
    fun create(): Electrician {
        val name = randomName()
        val experienceYear = randomAnno()
        val salary = randomSalary()
        val dailyHours = randomDailyHours()
        return Electrician(name, experienceYear, salary.toDouble(), dailyHours, BossAssigment.WITHOUT_BOSS)
    }
}